File: 283.cc
// Freddy @BH
// Jan 30, 2018


class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        while (true) {
            bool flag = true;
            int s = nums.size();
            for (int i=0;i<s-1;i++) {
                if(nums[i]==0 && nums[i+1]!=0) {
                    std::swap(nums[i],nums[i+1]);
                    flag=false;
                }
            }
            if (flag) break;
        }
    }
};

File: 283.py
# Freddy/Fangyu
# 12/18/2016

# Freddy @BH
# 1/30/2018

class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        aws=[]
        for num in nums:
            if num!=0:
                aws.append(num)
        for i in range(nums.count(0)):
            aws.append(0)
        for i in range(len(nums)):
            nums[i]=aws[i]
        
